In [1]:
import pandas as pd
import numpy as np
pd.set_option("display.max_columns", None)
pd.set_option("display.max_rows", None)
df = pd.read_excel("ASGDEUR5.xlsx")
df.info()
print(df.columns.tolist())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4611 entries, 0 to 4610
Columns: 155 entries, IDCNTRY to SCOPE
dtypes: float64(129), int64(26)
memory usage: 5.5 MB
['IDCNTRY', 'IDPOP', 'IDGRADER', 'IDGRADE', 'WAVE', 'IDSCHOOL', 'IDCLASS', 'IDSTUD', 'ITSEX', 'ITADMINI', 'ITLANG_SA', 'LCID_SA', 'ITLANG_SQ', 'LCID_SQ', 'IDBOOK', 'ASBG01', 'ASBG03', 'ASBG04', 'ASBG05A', 'ASBG05B', 'ASBG05C', 'ASBG05D', 'ASBG05E', 'ASBG05F', 'ASBG05G', 'ASBG05H', 'ASBG05I', 'ASBG05J', 'ASBG05K', 'ASBG06', 'ASBG07A', 'ASBG07B', 'ASBG08A', 'ASBG08B', 'ASBG09A', 'ASBG09B', 'ASBG09C', 'ASBG09D', 'ASBG09E', 'ASBG09F', 'ASBG09G', 'ASBG09H', 'ASBG10A', 'ASBG10B', 'ASBG10C', 'ASBG10D', 'ASBG10E', 'ASBG10F', 'ASBG11A', 'ASBG11B', 'ASBG11C', 'ASBG11D', 'ASBG11E', 'ASBG11F', 'ASBG11G', 'ASBG11H', 'ASBG11I', 'ASBG11J', 'ASBR01A', 'ASBR01B', 'ASBR01C', 'ASBR01D', 'ASBR01E', 'ASBR01F', 'ASBR01G', 'ASBR01H', 'ASBR01I', 'ASBR02A', 'ASBR02B', 'ASBR02C', 'ASBR02D', 'ASBR02E', 'ASBR03A', 'ASBR03B', 'ASBR03C', 'ASBR04', 'ASB

In [4]:
immigrant_school_IDs = [5011,
 5013,
 5016,
 5018,
 5025,
 5026,
 5033,
 5036,
 5040,
 5062,
 5075,
 5078,
 5081,
 5084,
 5087,
 5090,
 5099,
 5101,
 5106,
 5108,
 5123,
 5134,
 5137,
 5140,
 5144,
 5149,
 5156,
 5157,
 5158,
 5159,
 5161,
 5163,
 5164,
 5169,
 5170,
 5171,
 5190,
 5191,
 5193,
 5196,
 5205,
 5207,
 5209,
 5210,
 5219,
 5222,
 5229,
 5230,
 5231,
 5237]

We received the list above by filtering schools with high populations of immigrant children from the school context data. Now we will filter this dataframe so that we are only looking at the pupils from those schools. 

In [9]:
filtered_df = df[df['IDSCHOOL'].isin(immigrant_school_IDs)]

In [10]:
filtered_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 888 entries, 151 to 4027
Columns: 155 entries, IDCNTRY to SCOPE
dtypes: float64(129), int64(26)
memory usage: 1.1 MB


In [11]:
filtered_df["IDSCHOOL"].value_counts()

5209    25
5229    25
5026    24
5196    24
5090    24
5018    24
5087    23
5106    23
5011    23
5013    23
5219    22
5210    22
5149    22
5016    22
5101    22
5137    22
5075    22
5230    21
5062    20
5033    19
5140    19
5170    19
5231    19
5040    19
5078    18
5193    18
5084    17
5134    17
5161    17
5164    17
5191    17
5156    16
5171    16
5157    15
5123    15
5163    14
5036    14
5081    14
5158    13
5144    13
5190    12
5169    12
5222    12
5025    12
5099    12
5205    10
5207    10
5237    10
5108    10
5159     9
Name: IDSCHOOL, dtype: int64

The above is helpful as it tells us how many students we have from each school. We also know that we are looking at 888 students, some of whom will be immigrants and all of whom are attending schools with high immigrant populations. Now we can look a bit deeper into the demographics for these children. 

So let's first get a list of the student IDs for these kids.

In [12]:
filtered_df['IDSTUD'].to_list()

[50110301,
 50110302,
 50110303,
 50110304,
 50110305,
 50110306,
 50110307,
 50110308,
 50110309,
 50110310,
 50110311,
 50110312,
 50110313,
 50110314,
 50110315,
 50110316,
 50110317,
 50110318,
 50110319,
 50110320,
 50110321,
 50110322,
 50110323,
 50130101,
 50130102,
 50130103,
 50130104,
 50130105,
 50130106,
 50130107,
 50130108,
 50130110,
 50130111,
 50130112,
 50130113,
 50130114,
 50130115,
 50130116,
 50130117,
 50130118,
 50130119,
 50130120,
 50130121,
 50130122,
 50130123,
 50130124,
 50160201,
 50160202,
 50160203,
 50160204,
 50160205,
 50160206,
 50160207,
 50160208,
 50160209,
 50160210,
 50160211,
 50160212,
 50160213,
 50160214,
 50160215,
 50160216,
 50160218,
 50160220,
 50160221,
 50160222,
 50160223,
 50160224,
 50180201,
 50180202,
 50180203,
 50180204,
 50180205,
 50180206,
 50180207,
 50180208,
 50180209,
 50180210,
 50180211,
 50180212,
 50180213,
 50180214,
 50180215,
 50180216,
 50180217,
 50180219,
 50180221,
 50180222,
 50180223,
 50180224,
 50180225,

These appear to be made up of the school id and some other number. The fifth and sixth digits might pertain to class?

In [13]:
filtered_df['IDCLASS'].to_list()

[501103,
 501103,
 501103,
 501103,
 501103,
 501103,
 501103,
 501103,
 501103,
 501103,
 501103,
 501103,
 501103,
 501103,
 501103,
 501103,
 501103,
 501103,
 501103,
 501103,
 501103,
 501103,
 501103,
 501301,
 501301,
 501301,
 501301,
 501301,
 501301,
 501301,
 501301,
 501301,
 501301,
 501301,
 501301,
 501301,
 501301,
 501301,
 501301,
 501301,
 501301,
 501301,
 501301,
 501301,
 501301,
 501301,
 501602,
 501602,
 501602,
 501602,
 501602,
 501602,
 501602,
 501602,
 501602,
 501602,
 501602,
 501602,
 501602,
 501602,
 501602,
 501602,
 501602,
 501602,
 501602,
 501602,
 501602,
 501602,
 501802,
 501802,
 501802,
 501802,
 501802,
 501802,
 501802,
 501802,
 501802,
 501802,
 501802,
 501802,
 501802,
 501802,
 501802,
 501802,
 501802,
 501802,
 501802,
 501802,
 501802,
 501802,
 501802,
 501802,
 502502,
 502502,
 502502,
 502502,
 502502,
 502502,
 502502,
 502502,
 502502,
 502502,
 502502,
 502502,
 502601,
 502601,
 502601,
 502601,
 502601,
 502601,
 502601,
 

Yes, they do!

Now we will look at the Home Context Data to identify our Arabic speakers...